From 511315971f227ac2ce1481ff9e7350a607f78d23 Mon Sep 17 00:00:00 2001
From: Sam James <sam@gentoo.org>
Date: Sun, 23 Apr 2023 06:27:20 +0100
Subject: [PATCH] configure.ac: Silence harmless
 -Wimplicit-function-declaration warning in hpux sendfile check

We're currently looking for build systems where -Wimplicit-function-declaration
is emitted by checks as it often implies a missing include, but in this case,
it's fine - HPUX doesn't have a sendfile header, so it's just noise (and it
won't be true on Linux anyway).

But the check is run unconditionally and we already check if sendfile.h exists,
so include the header conditionally if it exists, as it's harmless and it avoids
it looking like there's a problem in pure-ftpd.

Unfortunately, it's not so easy to silence the sendfilev check which has
a similar (but different) problem because it uses AC_CHECK_FUNCS.

Bug: https://bugs.gentoo.org/900068
---
 configure.ac | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/configure.ac b/configure.ac
index 62768c8..0abb423 100644
--- a/configure.ac
+++ b/configure.ac
@@ -718,6 +718,9 @@ AC_MSG_CHECKING([if a hpuxish sendfile is available])
 AC_LINK_IFELSE([AC_LANG_PROGRAM([[
 #include <sys/socket.h>
 #include <stdio.h>
+#ifdef HAVE_SYS_SENDFILE_H
+# include <sys/sendfile.h>
+#endif
 ]], [[
 do {
  int fd = 0;
